### PR TITLE
General improvements

### PR DESCRIPTION
**Pull request by fcatus**
Linted the repo and made general fixes and performance improvements.

### CODE REVIEW

File: pdfreader/types/xref.py [hunk 1]
```diff
@@ -41,7 +43,8 @@ class XRefEntry(BaseXrefEntry):
         No compressed objects support
     """
     def __init__(self, offset, number, generation, typ):
-        assert isinstance(offset, int) and offset >=0
```
Comment (SanJJ1) [inline pdfreader/types/xref.py:44]: don't agree with this. assert statements are more pythonic and serve the same purpose. More readable, more compact, no performance drawback.

File: pdfreader/codecs/decoder.py [hunk 2]
```diff
@@ -14,7 +14,7 @@
 
 
 class PredefinedCmaps(object):
-    _cache = dict()
```
Comment (SanJJ1) [inline pdfreader/codecs/decoder.py:17]: I can see the benefit of these changes. dict() is slower than {}, and less pythonic.

File: pdfreader/types/xref.py (hunk 1 above)
Comment (fcatus) [reply pdfreader/types/xref.py:44]: Reverted this back to the original lines.  Thanks for the input!

**Comment by fcatus** [comment]
> I approve of all the changes in this pull request, except the changes regarding the assert statement.

I resolved this and committed to the PR.

**Pull request merged**